### ISSUE TITLE
Aerosol diagnostic discrepancy: AEROD_v/EXTINCT account for prescribed volcanic aerosols, AODVIS does not.

### ISSUE THREAD

**Issue by hbrown18** [E3SM-Project/E3SM#4800]
Hello, 

I wanted to submit a comment regarding a discrepancy between aerosol diagnostic output in E3SMv2. The main concern is in regards to the variables AEROD_v, EXTINCT, and AODVIS. Aerosol optical depth (AOD) is initially calculated in the modal aerosol module as AODVIS (modal_aero_sw subroutine), which does not take into account prescribed volcanic aerosol in the stratosphere. A separate subroutine (aer_rad_props_sw) then calls this AOD value and overwrites stratospheric values with the prescribed volcanic aerosol, renaming it as AEROD_v. Furthermore, in the modal_aero_sw module, the aerosol extinction coefficient (EXTINCT) is updated to include prescribed volcanic aerosol. This means that AODVIS and EXTINCT, which are both output from the modal_aer_opt.F90, are not reporting the same atmospheric aerosol conditions.

The following figure shows a 1991 time series of AEROD_v, AODVIS, and AOD calculated offline by column integration of EXTINCT. Mt. Pinatubo erupted in early June which is represented by prescribed volcanic aerosol and is reflected in AEROD_v and AOD from EXTINCT, but not AODVIS.
<img width="626" alt="Screen Shot 2022-02-17 at 3 36 18 PM" src="https://user-images.githubusercontent.com/48843542/154582809-860d44ea-1977-42ac-90b7-2154e1d467d9.png">

I'm not sure the best way to fix this. One possibility seems to be adding an EXTINCT term to the aer_rad_props_sw subroutine that calculates AEROD_v and calling it something different than EXTINCT (i.e., EXTINCT_v). However, prescribed volcanic aerosols, which are accounted for in the radiation model (as well as AEROD_v and EXTINCT), are not represented in other modal_aer_opt.F90 diagnostics which could lead to misinterpretation of aerosol optics model output.

**Comment by singhbalwinder** [comment]
Thanks @hbrown18 for reporting this. I will look into it. Tagging @hwangacme to know if he has any suggestions.

**Comment by hwangacme** [comment]
@hbrown18 @singhbalwinder 
AODVIS is the diagnostic variable we often use as the total aerosol optical depth (in the visible band) contributed by the actual aerosol species, including those originating from natural and anthropogenic sources, represented by the MAM4 scheme. Explosive volcanic emissions are prescribed into the simulations directly as extinction coefficient. I assume that EXTINCT is the variable used by radiation schemes that need to consider both MAM4 aerosols and the explosive volcanic aerosol forcing. AEROD_v is more comparable to other models or satellite retrievals that include contribution from volcanic aerosols.

**Comment by hbrown18** [comment]
Thanks @hwangacme. I see what you are saying about AEROD_v. I don't think EXTINCT is used in the radiation scheme. I'm pretty sure that the radiation scheme uses AOD (tau) passed from the aer_rad_props.F90 module. My main concern is that EXTINCT, which I have always assumed can be integrated to get AODVIS, will actually give you AEROD_v. This difference will be quite large in the presence of explosive volcanic eruptions. Maybe I am missing a purpose for the inclusion of explosive volcanic eruptions in the variable EXTINCT, but I am inclined to think that this should not be included in modal_aer_opt.F90.

**Comment by hwangacme** [comment]
@hbrown18 
The tau used for radiation code should be different from any of these 2D diagnostic variables for AOD. It must have a tau value for each model layer, which comes from extinction coefficient and layer thickness. The extinction coefficient variable will have to include the prescribed contribution of volcanic eruptions.

**Comment by hbrown18** [comment]
@hwangacme 
The radiation diagnostic tau value you mention is used in aer_rad_props.F90 to calculate AEROD_v and the other radiation diagnostics, but the EXTINCT variable (from modal_aer_opt.F90) does not have any connection to the layer resolved tau used by the radiation code. EXTINCT should represent only natural and anthropogenic aerosol extinction (based on its inclusion in modal_aer_opt.F90), but instead it is overwritten to include the stratospheric influence of prescribed volcanic aerosol forcing before being output by modal_aer_opt.F90. I think that this is the main inconsistency in the code.

**Comment by hbrown18** [comment]
Ok, sounds good @hwangacme. Thanks for the discussion!

**Comment by rljacob** [comment]
How can the variables be documented better?  Metadata in the history file?  Comments in the code?  Change the names?

**Comment by hwangacme** [comment]
@rljacob 
In my opinion, probably both in the metadata (accurate longname of the variables) and comments in the code, but the metadata is more important, as users don't necessarily read the code.

**Comment by hbrown18** [comment]
@rljacob 
I agree with @hwangacme. I think having an updated longname in the metadata would be the most helpful as the model output is the first place I look. One thought for AEROD_v could be something like "Total Aerosol Optical Depth in visible band including contributions from volcanic aerosol".

**Comment by susburrows** [comment]
FYI @mahf708

**Comment by hwangacme** [comment]
@mahf708 
Please make sure AODALL is indeed identical to AODVIS. There might be two potential traps:
1) Component AODs estimated from speciated mass distributions are not necessarily additive.
2) Some AODs are only estimated for daytime (and the visible band).
